### ISSUE TITLE
Implement plugin binaries version control

### ISSUE THREAD

**Issue by Southclaws** [Southclaws/sampctl#36]
The last piece of the puzzle for full version control, package management and package runtime.

You can create a package, grab the dependencies with `sampctl project ensure` and run the code to execute tests, check outputs, or quickly spin up a server to test in-game. Unfortunately it's impossible to load plugins into the runtime as there's no way to declare a plugin binary as a dependency for a project *runtime* only the Pawn includes for the *build* stage.

This should be implemented as a method of declaring plugins in `samp.json` so a user could use a DependencyString style path instead of just a filename:

```json
{
	"plugins": [
		"BigETI/pawn-memory:2.0"
	]
}
```

And then, a `runtime` section should be added to `pawn.json` spec to allow passing a server configuration to the package runtime, which would allow the specification of plugin versions.

```json
{
	"entry": "sometest.pwn",
	"runtime": {
		"plugins": [
			"BigETI/pawn-memory:2.0"
		]
	}
}
```

**Comment by Southclaws** [comment]
Some more thoughts on this:

This should be implemented as part of the `server` package. There should be an `Ensure` equivalent for server runtimes now which does the same as Pawn package ensuring - downloads any necessary dependencies.

How this will work is slightly different though, the actual ensure stage needs to grab the `pawn.json` file from the repo, cloning would be a bit too much for this so it should use the GitHub API to grab the file via the `contents` endpoint.

Once the `pawn.json` file has been loaded into memory, the `resources` section can be read so we know where to grab the binaries from. It can then construct GitHub release URLs (similar to the compiler I guess, could even re-use code here) and download the files. If the files are archives, they can be extracted and poked around to find the necessary files based on the `resources.plugins` field.

Now I want to write most of this to be as re-usable as possible and decoupled from the concept of server runtimes - why? Because, as with MySQL, it's not only plugin binaries that exist in release archives - the rook build process may need includes and it would use the exact same process here, except looking at `resource.includes` instead of `resource.plugins`.

It's quite a bit of work but once this is done, 1.5.x will mark the full implementation of package management! 🎉

**Comment by Southclaws** [comment]
implemented in 1.5.0
